wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubApiClient.java

```diff
@@ -163,14 +208,21 @@ public String[] getTokenScopes(String authenticationToken)
   }
 
   private HttpRequest buildGithubApiRequest(URI uri, String authenticationToken) {
-    return HttpRequest.newBuilder(uri)
-        .headers(
-            "Authorization",
-            "token " + authenticationToken,
-            "Accept",
-            "application/vnd.github.v3+json")
-        .timeout(DEFAULT_HTTP_TIMEOUT)
-        .build();
+    if (authenticationToken == null) {
```

Comment:
Why do we need the check? Won't the empty authorisation header work?

Reply:
Server responds with 401 when the request contains header
```
Authorization: token null
```

Reply:
Will the GitHub API even work without the authorisation?

Reply:
Here is nothing changed.
If the method is called where authorization==null, it's not necessary to set uncompleted header.

Reply:
`null` is not expected here, so why do we need the change?

Reply:
It's always possible to get the last commit for public repository.

Reply:
BTW, I see it first [tries to fetch authentication token](https://github.com/eclipse-che/che-server/blob/main/wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AuthorizingFileContentProvider.java#L68) to load the file, but fails with UnknownScmProviderException and then tries to [get the file without authentication](https://github.com/eclipse-che/che-server/blob/main/wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AuthorizingFileContentProvider.java#L76).
So, accessing the github without authentication token is real use case.

Reply:
It doesn't mean that unauthorised API requests are sent. It just checks the retrieved token by testing a github user request.
According to the GitHub documentation GitHub API can accept some unauthorised requests: https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api#authenticating, so I would agree to keep it, but please
* Mark the `authenticationToken` parameter as `@Nullable`.
* Add the related documentation link to the methods description.

Reply:
According to https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api#authenticating GitHub API can accept some unauthorised requests, so I would agree to keep it.

Reply:
```JAVA
private HttpRequest buildGithubApiRequest(URI uri, @Nullable String authenticationToken) {
    String[] headers = {"Accept", "application/vnd.github.v3+json"};
    if (!isNullOrEmpty(authenticationToken)) {
      headers[2] = "Authorization";
      headers[3] = "token " + authenticationToken;
    }
    return HttpRequest.newBuilder(uri)
            .headers(headers)
            .timeout(DEFAULT_HTTP_TIMEOUT)
            .build();
  }
```

Reply:
Please Add the documentation link to the methods description to explain why the token is `@Nullable`

Reply:
Done